tests/mpmc_channel.rs

```diff
@@ -368,6 +387,38 @@ macro_rules! gen_mpmc_tests {
                 assert!(fut2.is_terminated());
             }
 
+            #[test]
+            fn stream_smoke_test() {
+                let channel = ChannelType::new();
+                let (waker, count) = new_count_waker();
+                let cx = &mut Context::from_waker(&waker);
+
+                let stream = channel.stream();
+                pin_mut!(stream);
+
+                let fut = stream.next();
```

Comment:
This tests through the `StreamExt` trait, right? Maybe it would be better to directly check the lower level `Stream` APIs in order to exclude the behavior from the extension functions from the test.

Reply:
Agreed, I'm just not familiar with it.